README.md

```diff
@@ -158,6 +158,11 @@ markers.on('clusterclick', function (a) {
 #### Bulk adding and removing Markers
 `addLayers` and `removeLayers` are bulk methods for adding and removing markers and should be favoured over the single versions when doing bulk addition/removal of markers. Each takes an array of markers. You can use [dedicated options](#chunked-addlayers) to fine-tune the behaviour of `addLayers`.
 
+These methods extract non-group layer children from Layer Group types, even deeply nested. _However_, be noted that:
+- The input array is modified. If you need to keep it untouched, clone it before passing it (e.g. `newArray = originalArray.slice()`).
```

Comment:
Hi,

Not sure I am clear about what you mean by "scanning and cloning", but an interesting workaround (for the modified input array) would be indeed to clone the array if we find a group.

I think it should not be difficult then to clone it during the main loop (and to do it only once, using a flag to prevent further cloning). That way, in "normal" situations where no group is passed, `addLayers` would behave exactly the same (except for the check for `instanceof L.LayerGroup` at each item, but this cannot be avoided). And if there is at least one group, the input array is cloned at the moment we find the first one.

What do you think?